.github/workflows/check-build-test.yml

```diff
@@ -62,17 +71,83 @@ jobs:
           name: ${{ github.job }}
           path: ${{ github.workspace }}/ccache-debug
 
+  sonar:
+    name: "Static Analysis"
+    runs-on: ubuntu-latest
+    container: docker.io/ogdf/clang:15
+    env:
+      CCACHE_COMPILERCHECK: "%compiler% -v"
+    steps:
+      - name: "Add workspace as a safe directory in containers"
+        run: git config --system --add safe.directory $GITHUB_WORKSPACE
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # fetch full history for blame info
+      - name: Install tools
+        run: |
+          apt-get update
+          apt-get install -y curl gcovr
+          pip install git+https://github.com/N-Coder/clang-tidy-converter.git
```

Comment:
Could this become a problem because of the pip/pipx change? But I guess we just fix that if it comes up.

Reply:
I'm not aware of any pip/pipx change, do you have details on that?

Reply:
With some newer versions of linux distributions, installing python applications via pip [only throws an error](https://stackoverflow.com/questions/75608323/how-do-i-solve-error-externally-managed-environment-every-time-i-use-pip-3), also see [here](https://peps.python.org/pep-0668/). It is advised to use pipx instead to always install/use applications in their own isolated environment (but that is not always available in e.g. older ubuntu-images). I'm not sure whether this would affect us here, but I'll just deal with it when updating the docker images.

Reply:
Ah, that thing. While the motivation for that change is sensible on user systems, it doesn't make sense in containers. Still, the ubuntu container is probably going to suffer from this at some point because it probably uses the same packages. The easiest fix might be removing the marker file that enables this mechanism in the Dockerfile build.

`pipx` will not work in this case, because the clang-tidy-converter repo I forked is not installing an executable but only a package (which can be invoked via `python -m clang_tidy_converter`). So the super-clean way would be fixing that, installing pipx and then using pipx. Still, a usable pip without resorting to `venv`s is nice within containers, so I'd go with the easier way of removing the marker file before we update the container.

Reply:
> I'd go with the easier way of removing the marker file before we update the container.

Yup, I'll do that :+1: